cmd/kubeadm/app/util/endpoint_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+)
+
+func TestGenerateMasterEndpoint(t *testing.T) {
+	var tests = []struct {
+		address  string
+		port     int32
+		endpoint string
+		expected bool
+	}{
+		{
```

Comment:
The second and fourth test cases are redundant, IMO. The first and third tests verify that with the specified input, you get the expected output. The second and fourth test cases check that with the specified input you don't get unexpected output. They should be removed and the expected flag is not needed.